### PR TITLE
refactor(ui): refactor cache sets table to use new utils

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Refactored cache sets table to no longer use the `separateStorageData` util.
- Removed old utils file and replaced with new one

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the storage tab of a machine that has cache sets (or go to a ready/allocated machine in he angular version and add a cache set from the available disks table)
- Check that the data is the same as before

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2281

### CODE REVIEW

File: ui/src/app/machines/views/MachineDetails/MachineStorage/CacheSetsTable/CacheSetsTable.tsx [hunk 1]
```diff
@@ -1,87 +1,54 @@
...
-const CacheSetsTable = ({ cacheSets }: Props): JSX.Element => {
-  const { currentSort, sortRows, updateSort } = useTableSort(getSortValue, {
-    key: "name",
-    direction: "descending",
-  });
-  // TODO: update useTableSort to TS with generics
-  // https://github.com/canonical-web-and-design/maas-ui/issues/1869
-  const sortedCacheSets = sortRows(cacheSets) as StorageDevice[];
+  if (machine && "disks" in machine) {
+    const rows: TSFixMe[] = [];
```
Comment (huwshimi) [inline ui/src/app/machines/views/MachineDetails/MachineStorage/CacheSetsTable/CacheSetsTable.tsx:19]: Is setting the type to `TSFixMe[]` for strict mode? If so could you use a reduce function instead to generate the rows?

**Review by huwshimi — APPROVED** [review]
One comment.

File: ui/src/app/machines/views/MachineDetails/MachineStorage/CacheSetsTable/CacheSetsTable.tsx (hunk 1 above)
Comment (Caleb-Ellis) [reply ui/src/app/machines/views/MachineDetails/MachineStorage/CacheSetsTable/CacheSetsTable.tsx:19]: Ah yeah I could! Thanks I'll make the change now

Comment (Caleb-Ellis) [reply ui/src/app/machines/views/MachineDetails/MachineStorage/CacheSetsTable/CacheSetsTable.tsx:19]: So using a reduce function doesn't solve strict mode completely - gives me the error `Cannot assign type 'X' to type 'never'` for all of the columns content. I still think using reduce is neater, but I'm leaving `TSFixMe[]`in as the generic for now.

**Pull request merged**